frontend/packages/console-shared/src/components/formik-fields/field-types.ts

```diff
@@ -16,25 +16,35 @@ export interface FieldProps {
   validated?: ValidatedOptions;
 }
 
-export interface InputFieldProps extends FieldProps {
-  type: TextInputTypes;
+export interface BaseInputFieldProps extends FieldProps {
   placeholder?: string;
   onChange?: (event) => void;
   onBlur?: (event) => void;
 }
 
+export interface GroupInputProps {
+  beforeInput?: React.ReactNode;
+  afterInput?: React.ReactNode;
+  textType?: TextType;
+}
+
 export interface TextAreaProps extends FieldProps {
   placeholder?: string;
   onChange?: (event) => void;
   onBlur?: (event) => void;
 }
 
+export enum TextType {
+  TextInput = 'text',
+  TextArea = 'textArea',
+}
+
 export interface CheckboxFieldProps extends FieldProps {
   formLabel?: string;
   value?: string;
 }
 
-export interface SearchInputFieldProps extends InputFieldProps {
+export interface SearchInputFieldProps extends BaseInputFieldProps {
```

Comment:
This will also need a `type` prop but you've removed it from the base type interface.